frontend/packages/console-dynamic-plugin-sdk/src/utils/k8s/hooks/useK8sWatchResources.ts

```diff
@@ -65,7 +63,7 @@ export const useK8sWatchResources: UseK8sWatchResources = (initResources) => {
     k8sModelsRef.current = allK8sModels.filter(
       (model) =>
         requiredModels.includes(getReferenceForModel(model)) || requiredModels.includes(model.kind),
-    );
+    ) as ImmutableMap<string, K8sModel>;
```

Comment:
Would be good to understand why `as` is needed here. It's not obvious to me looking at the code.

But I definitely don't want to hold the PR on this.

Reply:
Fixed. There is a naming collision between the native `Iterable` and Immutable JS `Iterable` types which caused confusion.